### ISSUE TITLE
`./Setup configure --flags= ...` fails with ghc-8.2.2-rc1

### ISSUE THREAD

**Issue by duog** [haskell/cabal#4808]
The cabal packaged with ghc-8.2.2-rc1 (Cabal commit: c84a3c72196d0a5361f8ab77c6d8cb63b7a5d55d) fails when passed Setup configure is passed "--flags=".

See a log here:
https://gist.github.com/duog/260589ccbf63200addbea00f25436a48

**Comment by 23Skidoo** [comment]
@hvr just confirmed that this is a regression relative to 2.0.0.2.

**Comment by hvr** [comment]
@bgamari @23Skidoo  @phadej 

I've noticed this occurs on HEAD too; one way to workaround this is the following change to the parser:

```hs
diff --git a/Cabal/Distribution/Types/GenericPackageDescription.hs b/Cabal/Distribution/Types/GenericPackageDescription.hs
index 1547ade..a278370 100644
--- a/Cabal/Distribution/Types/GenericPackageDescription.hs
+++ b/Cabal/Distribution/Types/GenericPackageDescription.hs
@@ -159,7 +159,7 @@ dispFlagAssignment = Disp.hsep . map (Disp.text . showFlagValue)
 
 -- | Parses a flag assignment.
 parsecFlagAssignment :: ParsecParser FlagAssignment
-parsecFlagAssignment = P.sepBy1 (onFlag <|> offFlag) P.skipSpaces1
+parsecFlagAssignment = P.sepBy (onFlag <|> offFlag) P.skipSpaces1
   where
     onFlag = do
         P.optional (P.char '+')
@@ -172,7 +172,7 @@ parsecFlagAssignment = P.sepBy1 (onFlag <|> offFlag) P.skipSpaces1
 
 -- | Parses a flag assignment.
 parseFlagAssignment :: Parse.ReadP r FlagAssignment
-parseFlagAssignment = Parse.sepBy1 parseFlagValue Parse.skipSpaces1
+parseFlagAssignment = Parse.sepBy parseFlagValue Parse.skipSpaces1
   where
     parseFlagValue =
           (do Parse.optional (Parse.char '+')
```

Fwiw, `dispFlagAssignment []` results in an empty output; so this would be consistent.

**Comment by 23Skidoo** [comment]
This was fixed by #4830.
